File: Analysis/Association_Analysis_Dissimilarity.py
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from scipy import stats

data = pd.read_excel('Source Data File.xlsx')


increased = data['Overall_Increased_Similarity']
dissimilarity = data['Overall_Microbiome_Dissimilarity']


df = pd.DataFrame({
    'Increased': increased,
    'Dissimilarity': dissimilarity
}).dropna()


corr_coef, p_value = stats.pearsonr(df['Increased'], df['Dissimilarity'])


plt.figure(figsize=(8, 6))


df['Color'] = np.where(df['Increased'] > 0, 'Positive', 'Negative')


sns.scatterplot(data=df, x='Increased', y='Dissimilarity', 
                hue='Color', palette={'Positive': 'blue', 'Negative': 'red'},
                s=40, alpha=0.7)


sns.regplot(data=df, x='Increased', y='Dissimilarity', 
            scatter=False, color='black', line_kws={'linestyle': '--'})


plt.text(0.05, 0.95, f'R = {corr_coef:.2f}\np = {p_value:.3f}',
         transform=plt.gca().transAxes,
         fontsize=10, verticalalignment='top',
         bbox=dict(boxstyle='round', facecolor='wheat', alpha=0.5))

plt.xlabel('Increased Similarity')
plt.ylabel('Profile Dissimilarity')
plt.title('Association Analysis')
plt.legend(title='Increased Similarity')
plt.grid(True, alpha=0.3)
plt.tight_layout()
plt.show()

File: Analysis/Association_Analysis_Dissimilarity.m
%Association Analysis (Profile Dissimilarity vs Increased Profile Similarity)
data = readtable('Source Data File.xlsx');
increased = data.Overall_Increased_Similarity; % Increased Similarity in Microbiome Profiles
%increased = data.Pathway_increased_similarity; % Increased Similarity in Pathway Profiles
%increased = data.KO_increased; % Increased Similarity in KO Profiles

dissimilarity = data.Overall_Microbiome_Dissimilarity; % Dissimilarity between donor pre-FMT recipient in microbiome profile
%dissimilarity = data.Pathway_dissimilarity; % Dissimilarity between donor pre-FMT recipient in Pathway profile
%dissimilarity = data.KO_dissimilarity; % Dissimilarity between donor pre-FMT recipient in KO profile

valid_idx = ~isnan(increased) & ~isnan(dissimilarity);
increased = increased(valid_idx);
dissimilarity = dissimilarity(valid_idx);

% coefficient calculation and analysis
[corr_coef, p_value] = corr(increased, dissimilarity);

positive_mask = increased > 0;
negative_mask = increased < 0;

% Visualization
figure;
hold on;
scatter_size = 40;

scatter(increased(positive_mask), dissimilarity(positive_mask),...
    scatter_size, 'b', 'filled');
scatter(increased(negative_mask), dissimilarity(negative_mask),...
    scatter_size, 'r', 'filled');

coefficients = polyfit(increased, dissimilarity, 1);
x_fit = linspace(min(increased), max(increased), 100);
y_fit = polyval(coefficients, x_fit);
plot(x_fit, y_fit, 'k--', 'LineWidth', 1.5)

text(0.05, 0.95,...
    sprintf('R = %.2f\np = %.3f', corr_coef, p_value),...
    'Units', 'normalized',...
    'FontSize', 10,...
    'VerticalAlignment', 'top');
